datasketches/src/bloom/sketch.rs

```diff
@@ -0,0 +1,732 @@
+    ///
+    /// Returns:
+    /// - `true`: Item was **possibly** inserted (or false positive)
+    /// - `false`: Item was **definitely not** inserted
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use datasketches::bloom::BloomFilterBuilder;
+    /// let mut filter = BloomFilterBuilder::with_accuracy(100, 0.01).build();
+    /// filter.insert("apple");
+    ///
+    /// assert!(filter.contains(&"apple")); // true - was inserted
+    /// assert!(!filter.contains(&"grape")); // false - never inserted (probably)
+    /// ```
+    pub fn contains<T: Hash>(&self, item: &T) -> bool {
+        if self.is_empty() {
+            return false;
+        }
+
+        let (h0, h1) = self.compute_hash(item);
+        self.check_bits(h0, h1)
+    }
+
+    /// Tests and inserts an item in a single operation.
+    ///
+    /// Returns whether the item was possibly already in the set before insertion.
+    /// This is more efficient than calling `contains()` then `insert()` separately.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use datasketches::bloom::BloomFilterBuilder;
+    /// let mut filter = BloomFilterBuilder::with_accuracy(100, 0.01).build();
+    ///
+    /// let was_present = filter.contains_and_insert(&"apple");
+    /// assert!(!was_present); // First insertion
+    ///
+    /// let was_present = filter.contains_and_insert(&"apple");
+    /// assert!(was_present); // Now it's in the set
+    /// ```
+    pub fn contains_and_insert<T: Hash>(&mut self, item: &T) -> bool {
+        let (h0, h1) = self.compute_hash(item);
+        let was_present = self.check_bits(h0, h1);
+        self.set_bits(h0, h1);
+        was_present
+    }
+
+    /// Inserts an item into the filter.
+    ///
+    /// After insertion, `contains(item)` will always return `true`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use datasketches::bloom::BloomFilterBuilder;
+    /// let mut filter = BloomFilterBuilder::with_accuracy(100, 0.01).build();
+    ///
+    /// filter.insert("apple");
+    /// filter.insert(42_u64);
+    /// filter.insert(&[1, 2, 3]);
+    ///
+    /// assert!(filter.contains(&"apple"));
+    /// ```
+    pub fn insert<T: Hash>(&mut self, item: T) {
+        let (h0, h1) = self.compute_hash(&item);
+        self.set_bits(h0, h1);
+    }
+
+    /// Resets the filter to its initial empty state.
+    ///
+    /// Clears all bits while preserving capacity and configuration.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use datasketches::bloom::BloomFilterBuilder;
+    /// let mut filter = BloomFilterBuilder::with_accuracy(100, 0.01).build();
+    /// filter.insert("apple");
+    /// assert!(!filter.is_empty());
+    ///
+    /// filter.reset();
+    /// assert!(filter.is_empty());
+    /// assert!(!filter.contains(&"apple"));
+    /// ```
+    pub fn reset(&mut self) {
+        for word in &mut self.bit_array {
+            *word = 0;
+        }
```

Comment:
```suggestion
        self.bit_array.fill(0);
```

Bikeshedding some code snippet :P

Reply:
Always appreciate learning about new rust API. Thanks!